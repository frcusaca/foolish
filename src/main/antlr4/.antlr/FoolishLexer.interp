token literal names:
null
'{'
'}'
'('
')'
';'
null
null
'='
'+'
'-'
'*'
'/'
'if'
'then'
'elif'
'else'
'???'
null
null
null
'\''

token symbolic names:
null
LBRACE
RBRACE
LPAREN
RPAREN
SEMI
LINE_COMMENT
BLOCK_COMMENT
ASSIGN
PLUS
MINUS
MUL
DIV
IF
THEN
ELIF
ELSE
UNKNOWN
INTEGER
WS
IDENTIFIER
APOSTROPHE

rule names:
LBRACE
RBRACE
LPAREN
RPAREN
SEMI
LINE_COMMENT
BLOCK_COMMENT
ASSIGN
PLUS
MINUS
MUL
DIV
IF
THEN
ELIF
ELSE
UNKNOWN
INTEGER
LETTER
DIGIT
UNDERSCORE
WS
IDENTIFIER
APOSTROPHE

channel names:
DEFAULT_TOKEN_CHANNEL
HIDDEN

mode names:
DEFAULT_MODE

atn:
[4, 0, 21, 144, 6, -1, 2, 0, 7, 0, 2, 1, 7, 1, 2, 2, 7, 2, 2, 3, 7, 3, 2, 4, 7, 4, 2, 5, 7, 5, 2, 6, 7, 6, 2, 7, 7, 7, 2, 8, 7, 8, 2, 9, 7, 9, 2, 10, 7, 10, 2, 11, 7, 11, 2, 12, 7, 12, 2, 13, 7, 13, 2, 14, 7, 14, 2, 15, 7, 15, 2, 16, 7, 16, 2, 17, 7, 17, 2, 18, 7, 18, 2, 19, 7, 19, 2, 20, 7, 20, 2, 21, 7, 21, 2, 22, 7, 22, 2, 23, 7, 23, 1, 0, 1, 0, 1, 1, 1, 1, 1, 2, 1, 2, 1, 3, 1, 3, 1, 4, 1, 4, 1, 5, 1, 5, 5, 5, 62, 8, 5, 10, 5, 12, 5, 65, 9, 5, 1, 5, 1, 5, 1, 6, 1, 6, 1, 6, 1, 6, 1, 6, 5, 6, 74, 8, 6, 10, 6, 12, 6, 77, 9, 6, 1, 6, 1, 6, 1, 6, 1, 6, 1, 6, 1, 7, 1, 7, 1, 8, 1, 8, 1, 9, 1, 9, 1, 10, 1, 10, 1, 11, 1, 11, 1, 12, 1, 12, 1, 12, 1, 13, 1, 13, 1, 13, 1, 13, 1, 13, 1, 14, 1, 14, 1, 14, 1, 14, 1, 14, 1, 15, 1, 15, 1, 15, 1, 15, 1, 15, 1, 16, 1, 16, 1, 16, 1, 16, 1, 17, 4, 17, 117, 8, 17, 11, 17, 12, 17, 118, 1, 18, 1, 18, 1, 19, 1, 19, 1, 20, 1, 20, 1, 21, 4, 21, 128, 8, 21, 11, 21, 12, 21, 129, 1, 21, 1, 21, 1, 22, 1, 22, 1, 22, 1, 22, 5, 22, 138, 8, 22, 10, 22, 12, 22, 141, 9, 22, 1, 23, 1, 23, 1, 75, 0, 24, 1, 1, 3, 2, 5, 3, 7, 4, 9, 5, 11, 6, 13, 7, 15, 8, 17, 9, 19, 10, 21, 11, 23, 12, 25, 13, 27, 14, 29, 15, 31, 16, 33, 17, 35, 18, 37, 0, 39, 0, 41, 0, 43, 19, 45, 20, 47, 21, 1, 0, 4, 2, 0, 10, 10, 13, 13, 2, 0, 65, 90, 97, 122, 1, 0, 48, 57, 3, 0, 9, 10, 13, 13, 32, 32, 148, 0, 1, 1, 0, 0, 0, 0, 3, 1, 0, 0, 0, 0, 5, 1, 0, 0, 0, 0, 7, 1, 0, 0, 0, 0, 9, 1, 0, 0, 0, 0, 11, 1, 0, 0, 0, 0, 13, 1, 0, 0, 0, 0, 15, 1, 0, 0, 0, 0, 17, 1, 0, 0, 0, 0, 19, 1, 0, 0, 0, 0, 21, 1, 0, 0, 0, 0, 23, 1, 0, 0, 0, 0, 25, 1, 0, 0, 0, 0, 27, 1, 0, 0, 0, 0, 29, 1, 0, 0, 0, 0, 31, 1, 0, 0, 0, 0, 33, 1, 0, 0, 0, 0, 35, 1, 0, 0, 0, 0, 43, 1, 0, 0, 0, 0, 45, 1, 0, 0, 0, 0, 47, 1, 0, 0, 0, 1, 49, 1, 0, 0, 0, 3, 51, 1, 0, 0, 0, 5, 53, 1, 0, 0, 0, 7, 55, 1, 0, 0, 0, 9, 57, 1, 0, 0, 0, 11, 59, 1, 0, 0, 0, 13, 68, 1, 0, 0, 0, 15, 83, 1, 0, 0, 0, 17, 85, 1, 0, 0, 0, 19, 87, 1, 0, 0, 0, 21, 89, 1, 0, 0, 0, 23, 91, 1, 0, 0, 0, 25, 93, 1, 0, 0, 0, 27, 96, 1, 0, 0, 0, 29, 101, 1, 0, 0, 0, 31, 106, 1, 0, 0, 0, 33, 111, 1, 0, 0, 0, 35, 116, 1, 0, 0, 0, 37, 120, 1, 0, 0, 0, 39, 122, 1, 0, 0, 0, 41, 124, 1, 0, 0, 0, 43, 127, 1, 0, 0, 0, 45, 133, 1, 0, 0, 0, 47, 142, 1, 0, 0, 0, 49, 50, 5, 123, 0, 0, 50, 2, 1, 0, 0, 0, 51, 52, 5, 125, 0, 0, 52, 4, 1, 0, 0, 0, 53, 54, 5, 40, 0, 0, 54, 6, 1, 0, 0, 0, 55, 56, 5, 41, 0, 0, 56, 8, 1, 0, 0, 0, 57, 58, 5, 59, 0, 0, 58, 10, 1, 0, 0, 0, 59, 63, 5, 33, 0, 0, 60, 62, 8, 0, 0, 0, 61, 60, 1, 0, 0, 0, 62, 65, 1, 0, 0, 0, 63, 61, 1, 0, 0, 0, 63, 64, 1, 0, 0, 0, 64, 66, 1, 0, 0, 0, 65, 63, 1, 0, 0, 0, 66, 67, 6, 5, 0, 0, 67, 12, 1, 0, 0, 0, 68, 69, 5, 33, 0, 0, 69, 70, 5, 33, 0, 0, 70, 75, 1, 0, 0, 0, 71, 74, 9, 0, 0, 0, 72, 74, 7, 0, 0, 0, 73, 71, 1, 0, 0, 0, 73, 72, 1, 0, 0, 0, 74, 77, 1, 0, 0, 0, 75, 76, 1, 0, 0, 0, 75, 73, 1, 0, 0, 0, 76, 78, 1, 0, 0, 0, 77, 75, 1, 0, 0, 0, 78, 79, 5, 33, 0, 0, 79, 80, 5, 33, 0, 0, 80, 81, 1, 0, 0, 0, 81, 82, 6, 6, 0, 0, 82, 14, 1, 0, 0, 0, 83, 84, 5, 61, 0, 0, 84, 16, 1, 0, 0, 0, 85, 86, 5, 43, 0, 0, 86, 18, 1, 0, 0, 0, 87, 88, 5, 45, 0, 0, 88, 20, 1, 0, 0, 0, 89, 90, 5, 42, 0, 0, 90, 22, 1, 0, 0, 0, 91, 92, 5, 47, 0, 0, 92, 24, 1, 0, 0, 0, 93, 94, 5, 105, 0, 0, 94, 95, 5, 102, 0, 0, 95, 26, 1, 0, 0, 0, 96, 97, 5, 116, 0, 0, 97, 98, 5, 104, 0, 0, 98, 99, 5, 101, 0, 0, 99, 100, 5, 110, 0, 0, 100, 28, 1, 0, 0, 0, 101, 102, 5, 101, 0, 0, 102, 103, 5, 108, 0, 0, 103, 104, 5, 105, 0, 0, 104, 105, 5, 102, 0, 0, 105, 30, 1, 0, 0, 0, 106, 107, 5, 101, 0, 0, 107, 108, 5, 108, 0, 0, 108, 109, 5, 115, 0, 0, 109, 110, 5, 101, 0, 0, 110, 32, 1, 0, 0, 0, 111, 112, 5, 63, 0, 0, 112, 113, 5, 63, 0, 0, 113, 114, 5, 63, 0, 0, 114, 34, 1, 0, 0, 0, 115, 117, 3, 39, 19, 0, 116, 115, 1, 0, 0, 0, 117, 118, 1, 0, 0, 0, 118, 116, 1, 0, 0, 0, 118, 119, 1, 0, 0, 0, 119, 36, 1, 0, 0, 0, 120, 121, 7, 1, 0, 0, 121, 38, 1, 0, 0, 0, 122, 123, 7, 2, 0, 0, 123, 40, 1, 0, 0, 0, 124, 125, 5, 95, 0, 0, 125, 42, 1, 0, 0, 0, 126, 128, 7, 3, 0, 0, 127, 126, 1, 0, 0, 0, 128, 129, 1, 0, 0, 0, 129, 127, 1, 0, 0, 0, 129, 130, 1, 0, 0, 0, 130, 131, 1, 0, 0, 0, 131, 132, 6, 21, 0, 0, 132, 44, 1, 0, 0, 0, 133, 139, 3, 37, 18, 0, 134, 138, 3, 37, 18, 0, 135, 138, 3, 39, 19, 0, 136, 138, 3, 41, 20, 0, 137, 134, 1, 0, 0, 0, 137, 135, 1, 0, 0, 0, 137, 136, 1, 0, 0, 0, 138, 141, 1, 0, 0, 0, 139, 137, 1, 0, 0, 0, 139, 140, 1, 0, 0, 0, 140, 46, 1, 0, 0, 0, 141, 139, 1, 0, 0, 0, 142, 143, 5, 39, 0, 0, 143, 48, 1, 0, 0, 0, 8, 0, 63, 73, 75, 118, 129, 137, 139, 1, 6, 0, 0]