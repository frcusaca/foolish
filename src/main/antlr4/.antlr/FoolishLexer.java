// Generated from /home/user/foolish/src/main/antlr4/Foolish.g4 by ANTLR 4.13.1
import org.antlr.v4.runtime.Lexer;
import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.Token;
import org.antlr.v4.runtime.TokenStream;
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.atn.*;
import org.antlr.v4.runtime.dfa.DFA;
import org.antlr.v4.runtime.misc.*;

@SuppressWarnings({"all", "warnings", "unchecked", "unused", "cast", "CheckReturnValue", "this-escape"})
public class FoolishLexer extends Lexer {
	static { RuntimeMetaData.checkVersion("4.13.1", RuntimeMetaData.VERSION); }

	protected static final DFA[] _decisionToDFA;
	protected static final PredictionContextCache _sharedContextCache =
		new PredictionContextCache();
	public static final int
		LBRACE=1, RBRACE=2, LPAREN=3, RPAREN=4, SEMI=5, LINE_COMMENT=6, BLOCK_COMMENT=7, 
		ASSIGN=8, PLUS=9, MINUS=10, MUL=11, DIV=12, IF=13, THEN=14, ELIF=15, ELSE=16, 
		UNKNOWN=17, INTEGER=18, WS=19, IDENTIFIER=20, APOSTROPHE=21;
	public static String[] channelNames = {
		"DEFAULT_TOKEN_CHANNEL", "HIDDEN"
	};

	public static String[] modeNames = {
		"DEFAULT_MODE"
	};

	private static String[] makeRuleNames() {
		return new String[] {
			"LBRACE", "RBRACE", "LPAREN", "RPAREN", "SEMI", "LINE_COMMENT", "BLOCK_COMMENT", 
			"ASSIGN", "PLUS", "MINUS", "MUL", "DIV", "IF", "THEN", "ELIF", "ELSE", 
			"UNKNOWN", "INTEGER", "LETTER", "DIGIT", "UNDERSCORE", "WS", "IDENTIFIER", 
			"APOSTROPHE"
		};
	}
	public static final String[] ruleNames = makeRuleNames();

	private static String[] makeLiteralNames() {
		return new String[] {
			null, "'{'", "'}'", "'('", "')'", "';'", null, null, "'='", "'+'", "'-'", 
			"'*'", "'/'", "'if'", "'then'", "'elif'", "'else'", "'???'", null, null, 
			null, "'''"
		};
	}
	private static final String[] _LITERAL_NAMES = makeLiteralNames();
	private static String[] makeSymbolicNames() {
		return new String[] {
			null, "LBRACE", "RBRACE", "LPAREN", "RPAREN", "SEMI", "LINE_COMMENT", 
			"BLOCK_COMMENT", "ASSIGN", "PLUS", "MINUS", "MUL", "DIV", "IF", "THEN", 
			"ELIF", "ELSE", "UNKNOWN", "INTEGER", "WS", "IDENTIFIER", "APOSTROPHE"
		};
	}
	private static final String[] _SYMBOLIC_NAMES = makeSymbolicNames();
	public static final Vocabulary VOCABULARY = new VocabularyImpl(_LITERAL_NAMES, _SYMBOLIC_NAMES);

	/**
	 * @deprecated Use {@link #VOCABULARY} instead.
	 */
	@Deprecated
	public static final String[] tokenNames;
	static {
		tokenNames = new String[_SYMBOLIC_NAMES.length];
		for (int i = 0; i < tokenNames.length; i++) {
			tokenNames[i] = VOCABULARY.getLiteralName(i);
			if (tokenNames[i] == null) {
				tokenNames[i] = VOCABULARY.getSymbolicName(i);
			}

			if (tokenNames[i] == null) {
				tokenNames[i] = "<INVALID>";
			}
		}
	}

	@Override
	@Deprecated
	public String[] getTokenNames() {
		return tokenNames;
	}

	@Override

	public Vocabulary getVocabulary() {
		return VOCABULARY;
	}


	public FoolishLexer(CharStream input) {
		super(input);
		_interp = new LexerATNSimulator(this,_ATN,_decisionToDFA,_sharedContextCache);
	}

	@Override
	public String getGrammarFileName() { return "Foolish.g4"; }

	@Override
	public String[] getRuleNames() { return ruleNames; }

	@Override
	public String getSerializedATN() { return _serializedATN; }

	@Override
	public String[] getChannelNames() { return channelNames; }

	@Override
	public String[] getModeNames() { return modeNames; }

	@Override
	public ATN getATN() { return _ATN; }

	public static final String _serializedATN =
		"\u0004\u0000\u0015\u0090\u0006\uffff\uffff\u0002\u0000\u0007\u0000\u0002"+
		"\u0001\u0007\u0001\u0002\u0002\u0007\u0002\u0002\u0003\u0007\u0003\u0002"+
		"\u0004\u0007\u0004\u0002\u0005\u0007\u0005\u0002\u0006\u0007\u0006\u0002"+
		"\u0007\u0007\u0007\u0002\b\u0007\b\u0002\t\u0007\t\u0002\n\u0007\n\u0002"+
		"\u000b\u0007\u000b\u0002\f\u0007\f\u0002\r\u0007\r\u0002\u000e\u0007\u000e"+
		"\u0002\u000f\u0007\u000f\u0002\u0010\u0007\u0010\u0002\u0011\u0007\u0011"+
		"\u0002\u0012\u0007\u0012\u0002\u0013\u0007\u0013\u0002\u0014\u0007\u0014"+
		"\u0002\u0015\u0007\u0015\u0002\u0016\u0007\u0016\u0002\u0017\u0007\u0017"+
		"\u0001\u0000\u0001\u0000\u0001\u0001\u0001\u0001\u0001\u0002\u0001\u0002"+
		"\u0001\u0003\u0001\u0003\u0001\u0004\u0001\u0004\u0001\u0005\u0001\u0005"+
		"\u0005\u0005>\b\u0005\n\u0005\f\u0005A\t\u0005\u0001\u0005\u0001\u0005"+
		"\u0001\u0006\u0001\u0006\u0001\u0006\u0001\u0006\u0001\u0006\u0005\u0006"+
		"J\b\u0006\n\u0006\f\u0006M\t\u0006\u0001\u0006\u0001\u0006\u0001\u0006"+
		"\u0001\u0006\u0001\u0006\u0001\u0007\u0001\u0007\u0001\b\u0001\b\u0001"+
		"\t\u0001\t\u0001\n\u0001\n\u0001\u000b\u0001\u000b\u0001\f\u0001\f\u0001"+
		"\f\u0001\r\u0001\r\u0001\r\u0001\r\u0001\r\u0001\u000e\u0001\u000e\u0001"+
		"\u000e\u0001\u000e\u0001\u000e\u0001\u000f\u0001\u000f\u0001\u000f\u0001"+
		"\u000f\u0001\u000f\u0001\u0010\u0001\u0010\u0001\u0010\u0001\u0010\u0001"+
		"\u0011\u0004\u0011u\b\u0011\u000b\u0011\f\u0011v\u0001\u0012\u0001\u0012"+
		"\u0001\u0013\u0001\u0013\u0001\u0014\u0001\u0014\u0001\u0015\u0004\u0015"+
		"\u0080\b\u0015\u000b\u0015\f\u0015\u0081\u0001\u0015\u0001\u0015\u0001"+
		"\u0016\u0001\u0016\u0001\u0016\u0001\u0016\u0005\u0016\u008a\b\u0016\n"+
		"\u0016\f\u0016\u008d\t\u0016\u0001\u0017\u0001\u0017\u0001K\u0000\u0018"+
		"\u0001\u0001\u0003\u0002\u0005\u0003\u0007\u0004\t\u0005\u000b\u0006\r"+
		"\u0007\u000f\b\u0011\t\u0013\n\u0015\u000b\u0017\f\u0019\r\u001b\u000e"+
		"\u001d\u000f\u001f\u0010!\u0011#\u0012%\u0000\'\u0000)\u0000+\u0013-\u0014"+
		"/\u0015\u0001\u0000\u0004\u0002\u0000\n\n\r\r\u0002\u0000AZaz\u0001\u0000"+
		"09\u0003\u0000\t\n\r\r  \u0094\u0000\u0001\u0001\u0000\u0000\u0000\u0000"+
		"\u0003\u0001\u0000\u0000\u0000\u0000\u0005\u0001\u0000\u0000\u0000\u0000"+
		"\u0007\u0001\u0000\u0000\u0000\u0000\t\u0001\u0000\u0000\u0000\u0000\u000b"+
		"\u0001\u0000\u0000\u0000\u0000\r\u0001\u0000\u0000\u0000\u0000\u000f\u0001"+
		"\u0000\u0000\u0000\u0000\u0011\u0001\u0000\u0000\u0000\u0000\u0013\u0001"+
		"\u0000\u0000\u0000\u0000\u0015\u0001\u0000\u0000\u0000\u0000\u0017\u0001"+
		"\u0000\u0000\u0000\u0000\u0019\u0001\u0000\u0000\u0000\u0000\u001b\u0001"+
		"\u0000\u0000\u0000\u0000\u001d\u0001\u0000\u0000\u0000\u0000\u001f\u0001"+
		"\u0000\u0000\u0000\u0000!\u0001\u0000\u0000\u0000\u0000#\u0001\u0000\u0000"+
		"\u0000\u0000+\u0001\u0000\u0000\u0000\u0000-\u0001\u0000\u0000\u0000\u0000"+
		"/\u0001\u0000\u0000\u0000\u00011\u0001\u0000\u0000\u0000\u00033\u0001"+
		"\u0000\u0000\u0000\u00055\u0001\u0000\u0000\u0000\u00077\u0001\u0000\u0000"+
		"\u0000\t9\u0001\u0000\u0000\u0000\u000b;\u0001\u0000\u0000\u0000\rD\u0001"+
		"\u0000\u0000\u0000\u000fS\u0001\u0000\u0000\u0000\u0011U\u0001\u0000\u0000"+
		"\u0000\u0013W\u0001\u0000\u0000\u0000\u0015Y\u0001\u0000\u0000\u0000\u0017"+
		"[\u0001\u0000\u0000\u0000\u0019]\u0001\u0000\u0000\u0000\u001b`\u0001"+
		"\u0000\u0000\u0000\u001de\u0001\u0000\u0000\u0000\u001fj\u0001\u0000\u0000"+
		"\u0000!o\u0001\u0000\u0000\u0000#t\u0001\u0000\u0000\u0000%x\u0001\u0000"+
		"\u0000\u0000\'z\u0001\u0000\u0000\u0000)|\u0001\u0000\u0000\u0000+\u007f"+
		"\u0001\u0000\u0000\u0000-\u0085\u0001\u0000\u0000\u0000/\u008e\u0001\u0000"+
		"\u0000\u000012\u0005{\u0000\u00002\u0002\u0001\u0000\u0000\u000034\u0005"+
		"}\u0000\u00004\u0004\u0001\u0000\u0000\u000056\u0005(\u0000\u00006\u0006"+
		"\u0001\u0000\u0000\u000078\u0005)\u0000\u00008\b\u0001\u0000\u0000\u0000"+
		"9:\u0005;\u0000\u0000:\n\u0001\u0000\u0000\u0000;?\u0005!\u0000\u0000"+
		"<>\b\u0000\u0000\u0000=<\u0001\u0000\u0000\u0000>A\u0001\u0000\u0000\u0000"+
		"?=\u0001\u0000\u0000\u0000?@\u0001\u0000\u0000\u0000@B\u0001\u0000\u0000"+
		"\u0000A?\u0001\u0000\u0000\u0000BC\u0006\u0005\u0000\u0000C\f\u0001\u0000"+
		"\u0000\u0000DE\u0005!\u0000\u0000EF\u0005!\u0000\u0000FK\u0001\u0000\u0000"+
		"\u0000GJ\t\u0000\u0000\u0000HJ\u0007\u0000\u0000\u0000IG\u0001\u0000\u0000"+
		"\u0000IH\u0001\u0000\u0000\u0000JM\u0001\u0000\u0000\u0000KL\u0001\u0000"+
		"\u0000\u0000KI\u0001\u0000\u0000\u0000LN\u0001\u0000\u0000\u0000MK\u0001"+
		"\u0000\u0000\u0000NO\u0005!\u0000\u0000OP\u0005!\u0000\u0000PQ\u0001\u0000"+
		"\u0000\u0000QR\u0006\u0006\u0000\u0000R\u000e\u0001\u0000\u0000\u0000"+
		"ST\u0005=\u0000\u0000T\u0010\u0001\u0000\u0000\u0000UV\u0005+\u0000\u0000"+
		"V\u0012\u0001\u0000\u0000\u0000WX\u0005-\u0000\u0000X\u0014\u0001\u0000"+
		"\u0000\u0000YZ\u0005*\u0000\u0000Z\u0016\u0001\u0000\u0000\u0000[\\\u0005"+
		"/\u0000\u0000\\\u0018\u0001\u0000\u0000\u0000]^\u0005i\u0000\u0000^_\u0005"+
		"f\u0000\u0000_\u001a\u0001\u0000\u0000\u0000`a\u0005t\u0000\u0000ab\u0005"+
		"h\u0000\u0000bc\u0005e\u0000\u0000cd\u0005n\u0000\u0000d\u001c\u0001\u0000"+
		"\u0000\u0000ef\u0005e\u0000\u0000fg\u0005l\u0000\u0000gh\u0005i\u0000"+
		"\u0000hi\u0005f\u0000\u0000i\u001e\u0001\u0000\u0000\u0000jk\u0005e\u0000"+
		"\u0000kl\u0005l\u0000\u0000lm\u0005s\u0000\u0000mn\u0005e\u0000\u0000"+
		"n \u0001\u0000\u0000\u0000op\u0005?\u0000\u0000pq\u0005?\u0000\u0000q"+
		"r\u0005?\u0000\u0000r\"\u0001\u0000\u0000\u0000su\u0003\'\u0013\u0000"+
		"ts\u0001\u0000\u0000\u0000uv\u0001\u0000\u0000\u0000vt\u0001\u0000\u0000"+
		"\u0000vw\u0001\u0000\u0000\u0000w$\u0001\u0000\u0000\u0000xy\u0007\u0001"+
		"\u0000\u0000y&\u0001\u0000\u0000\u0000z{\u0007\u0002\u0000\u0000{(\u0001"+
		"\u0000\u0000\u0000|}\u0005_\u0000\u0000}*\u0001\u0000\u0000\u0000~\u0080"+
		"\u0007\u0003\u0000\u0000\u007f~\u0001\u0000\u0000\u0000\u0080\u0081\u0001"+
		"\u0000\u0000\u0000\u0081\u007f\u0001\u0000\u0000\u0000\u0081\u0082\u0001"+
		"\u0000\u0000\u0000\u0082\u0083\u0001\u0000\u0000\u0000\u0083\u0084\u0006"+
		"\u0015\u0000\u0000\u0084,\u0001\u0000\u0000\u0000\u0085\u008b\u0003%\u0012"+
		"\u0000\u0086\u008a\u0003%\u0012\u0000\u0087\u008a\u0003\'\u0013\u0000"+
		"\u0088\u008a\u0003)\u0014\u0000\u0089\u0086\u0001\u0000\u0000\u0000\u0089"+
		"\u0087\u0001\u0000\u0000\u0000\u0089\u0088\u0001\u0000\u0000\u0000\u008a"+
		"\u008d\u0001\u0000\u0000\u0000\u008b\u0089\u0001\u0000\u0000\u0000\u008b"+
		"\u008c\u0001\u0000\u0000\u0000\u008c.\u0001\u0000\u0000\u0000\u008d\u008b"+
		"\u0001\u0000\u0000\u0000\u008e\u008f\u0005\'\u0000\u0000\u008f0\u0001"+
		"\u0000\u0000\u0000\b\u0000?IKv\u0081\u0089\u008b\u0001\u0006\u0000\u0000";
	public static final ATN _ATN =
		new ATNDeserializer().deserialize(_serializedATN.toCharArray());
	static {
		_decisionToDFA = new DFA[_ATN.getNumberOfDecisions()];
		for (int i = 0; i < _ATN.getNumberOfDecisions(); i++) {
			_decisionToDFA[i] = new DFA(_ATN.getDecisionState(i), i);
		}
	}
}