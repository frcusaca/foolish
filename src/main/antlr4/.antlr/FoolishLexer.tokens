LBRACE=1
RBRACE=2
LPAREN=3
RPAREN=4
SEMI=5
LINE_COMMENT=6
BLOCK_COMMENT=7
ASSIGN=8
PLUS=9
MINUS=10
MUL=11
DIV=12
IF=13
THEN=14
ELIF=15
ELSE=16
UNKNOWN=17
INTEGER=18
WS=19
IDENTIFIER=20
APOSTROPHE=21
'{'=1
'}'=2
'('=3
')'=4
';'=5
'='=8
'+'=9
'-'=10
'*'=11
'/'=12
'if'=13
'then'=14
'elif'=15
'else'=16
'???'=17
'\''=21
