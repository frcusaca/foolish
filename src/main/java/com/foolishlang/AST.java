
package com.foolishlang;

import java.util.*;

public final class AST {
    public sealed interface Node permits Program, Brane, Stmt, Expr, TypeExpr, Comment {}

    public static final class Program implements Node {
        public final Brane body;
        public Program(Brane body) { this.body = body; }
    }

    public static final class Brane implements Node {
        public final List<Stmt> statements;
        public Brane(List<Stmt> statements) { this.statements = statements; }
    }

    public sealed interface Stmt extends Node permits AssignStmt, TypeAssignStmt, DerefAssignStmt, ExprStmt, Comment {}

    public static final class AssignStmt implements Stmt {
        public final Expr id; // Identifier or TypeIdentifier
        public final Expr expr;
        public AssignStmt(Expr id, Expr expr) { this.id = id; this.expr = expr; }
    }

    public static final class TypeAssignStmt implements Stmt {
        public final TypeIdentifier id;
        public final Node typeExpr; // TypeExpr or TypeIdentifier
        public TypeAssignStmt(TypeIdentifier id, Node typeExpr) { this.id = id; this.typeExpr = typeExpr; }
    }

    public static final class DerefAssignStmt implements Stmt {
        public final Identifier id;
        public final Expr expr;
        public DerefAssignStmt(Identifier id, Expr expr) { this.id = id; this.expr = expr; }
    }

    public static final class ExprStmt implements Stmt {
        public final Expr expr;
        public ExprStmt(Expr expr) { this.expr = expr; }
    }

    public static final class Comment implements Stmt {
        public final String text;
        public Comment(String text) { this.text = text; }
    }

    public sealed interface Expr extends Node permits Identifier, TypeIdentifier, Literal, FuncExpr, BraneExpr, ConcatExpr, BinaryExpr, PathDerefExpr, ParenExpr {}

    public static final class ParenExpr implements Expr {
        public final Expr expr;
        public ParenExpr(Expr expr) { this.expr = expr; }
    }

    public static final class Identifier implements Expr {
        public final String name;
        public final boolean isOrdinaryQuoted;
        public Identifier(String name, boolean isOrdinaryQuoted) { this.name = name; this.isOrdinaryQuoted = isOrdinaryQuoted; }
        @Override public String toString(){ return "Id("+name+")"; }
    }

    public static final class TypeIdentifier implements Expr {
        public final String name;
        public TypeIdentifier(String name) { this.name = name; }
        @Override public String toString(){ return "TId("+name+")"; }
    }

    public sealed interface Literal extends Expr permits IntLiteral, FloatLiteral, StringLiteral, TypeLiteral {}

    public static final class IntLiteral implements Literal { public final long value; public IntLiteral(long v){ this.value=v; } }
    public static final class FloatLiteral implements Literal { public final double value; public FloatLiteral(double v){ this.value=v; } }
    public static final class StringLiteral implements Literal { public final String value; public StringLiteral(String v){ this.value=v; } }
    public static final class TypeLiteral implements Literal { public final Literal inner; public TypeLiteral(Literal inner){ this.inner=inner; } }

    public static final class FuncExpr implements Expr {
        public final List<Param> params;
        public final Brane body;
        public FuncExpr(List<Param> params, Brane body){ this.params=params; this.body=body; }
    }

    public static final class Param implements Node {
        public final Expr id; // Identifier or TypeIdentifier
        public final Node type; // TypeExpr or TypeIdentifier
        public Param(Expr id, Node type){ this.id=id; this.type=type; }
    }

    public sealed interface TypeExpr extends Node permits TypeRef, BraneType {}

    public static final class TypeRef implements TypeExpr {
        public final String name;
        public final boolean primitive;
        public TypeRef(String name, boolean primitive){ this.name=name; this.primitive=primitive; }
    }

    public static final class BraneType implements TypeExpr {
        public final List<FieldDef> fields;
        public BraneType(List<FieldDef> fields){ this.fields=fields; }
    }

    public static final class FieldDef implements Node {
        public final Expr id; // Identifier or TypeIdentifier
        public final Node type; // TypeExpr or TypeIdentifier
        public FieldDef(Expr id, Node type){ this.id=id; this.type=type; }
    }

    public static final class BraneExpr implements Expr {
        public final List<Stmt> statements;
        public BraneExpr(List<Stmt> statements){ this.statements=statements; }
    }

    public static final class ConcatExpr implements Expr {
        public final List<Expr> items; // left-to-right sequence (RPN style adjacency)
        public ConcatExpr(List<Expr> items){ this.items=items; }
    }

    public enum BinaryOp { OROR, ANDAND, EQEQ, LT, GT, LE, GE, PLUS, MINUS, STAR, SLASH }

    public static final class BinaryExpr implements Expr {
        public final BinaryOp op;
        public final Expr left, right;
        public BinaryExpr(BinaryOp op, Expr left, Expr right){ this.op=op; this.left=left; this.right=right; }
    }

    public static final class PathDeref implements Node {
        public final String op; // "^", "$", "#"
        public final Node index; // Identifier/IntLiteral or null
        public PathDeref(String op, Node index){ this.op=op; this.index=index; }
    }

    public static final class PathDerefExpr implements Expr {
        public final Expr base;
        public final List<PathDeref> derefs;
        public PathDerefExpr(Expr base, List<PathDeref> derefs){ this.base=base; this.derefs=derefs; }
    }
}
