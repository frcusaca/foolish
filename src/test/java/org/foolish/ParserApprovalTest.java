package org.foolish;

import org.foolish.grammar.FoolishLexer;
import org.foolish.grammar.FoolishParser;
import org.foolish.ast.AST;
import org.foolish.ast.ASTBuilder;
import org.antlr.v4.runtime.*;
import org.antlr.v4.runtime.tree.ParseTree;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

public class ParserApprovalTest {

    private AST parse(String code) {
        CharStream input = CharStreams.fromString(code);
        FoolishLexer lexer = new FoolishLexer(input);
        CommonTokenStream tokens = new CommonTokenStream(lexer);
        FoolishParser parser = new FoolishParser(tokens);
        ParseTree tree = parser.program();
        return new ASTBuilder().visit(tree);
    }

    @Test
    void arithmeticIsApproved() {
        AST ast = parse("{ x = 1+2*3; y = x-4; }");
        Approvals.verify(ast.toString());
    }
}
