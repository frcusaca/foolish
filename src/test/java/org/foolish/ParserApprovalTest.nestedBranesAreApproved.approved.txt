{
  {
  {
  z = 3;
};
  y = 2;
  {
  w = 4;
};
};
  x = 1;
  {
  p = 5;
  {
  q = 6;
};
};
}
